src/types/protocol.ts

```diff
@@ -142,12 +142,12 @@ export function isRosbridgeUnadvertiseServiceMessage(
   return message.op === "unadvertise_service";
 }
 
-export interface RosbridgeCallServiceMessage<TArgs = unknown[]>
+export interface RosbridgeCallServiceMessage<TArgs = void>
   extends RosbridgeMessage {
   op: "call_service";
   id?: string;
   service: string;
-  args?: TArgs;
```

Comment:
Rosbridge protocol description says `args` field is optional

Reply:
> Also `values` in `RosbridgeServiceResponseMessage` can be of type `string` when the `result` is set to false

That presents an interesting polymorphism issue.. I'll see if I can manage to address it without a major overhaul

Reply:
I don't think roslibjs should allow its consumers to send unrelated data types, but maybe it should allow Partials. The logic in rosbridge will deeply fill missing fields with defaults, right? Maybe this should be some sort of `DeepPartial<TArgs>`

Reply:
Yep, rosbridge fills missing fields with defaults

Reply:
As for passing Array as `args`. I don't think we need to support it in Typescript port.